src/redturtle/volto/restapi/deserializer/relationfield.py

```diff
@@ -0,0 +1,61 @@
+from plone.dexterity.interfaces import IDexterityContent
+from plone.restapi.deserializer.dxfields import DefaultFieldDeserializer
+from plone.restapi.interfaces import IFieldDeserializer
+from Products.CMFCore.utils import getToolByName
+from z3c.relationfield.interfaces import IRelationChoice
+from zope.component import adapter
+from zope.component import getMultiAdapter
+from zope.component import queryUtility
+from zope.interface import implementer
+from zope.intid.interfaces import IIntIds
+
+from redturtle.volto.interfaces import IRedturtleVoltoLayer
+
+
+@implementer(IFieldDeserializer)
+@adapter(IRelationChoice, IDexterityContent, IRedturtleVoltoLayer)
+class RelationChoiceFieldDeserializer(DefaultFieldDeserializer):
+    def __call__(self, value):
+        obj = None
+        resolved_by = None
+
+        if isinstance(value, dict):
+            # We are trying to deserialize the output of a serialization
+            # which is enhanced, extract it and put it on the loop again
+            value = value["UID"] if value.get("UID", None) else value["@id"]
+
+        if isinstance(value, int):
+            # Resolve by intid
+            intids = queryUtility(IIntIds)
+            obj = intids.queryObject(value)
+            resolved_by = "intid"
+        elif isinstance(value, str):
+            portal = getMultiAdapter(
+                (self.context, self.request), name="plone_portal_state"
+            ).portal()
+            portal_url = portal.absolute_url()
+            if value.startswith(portal_url):
+                # Resolve by URL
+                obj = portal.restrictedTraverse(value[len(portal_url) + 1:], None)
```

Comment:
Questa cosa è sbagliata. Si sta usando una url come fosse un path. Esistono eventualmente delle utility che convertono una url in un path rispettando eventuali configurazioni vhm

Reply:
sistemato

Reply:
Non mi sembra. Si sta ancora facendo traverse su una url (il path della url, ma la url). C'è differenza tra un path plone e una url. Va anche bene come soluzione temporanea ma aggiungiamo almeno un commento per segnalare che in futuro potrebbero esserci problemi qui.

Reply:
tipo? a che cosa stavi pensando?